### ISSUE TITLE
Kokkos::sort_by_key performance cliff without TPLS (rocThrust, thrust, etc)

### ISSUE THREAD

**Issue by cwpearson** [kokkos/kokkos-kernels#2352]
https://github.com/kokkos/kokkos-kernels/blob/101b8ae4d2d46e6f49ca71bcbde75816b52380c4/sparse/impl/KokkosSparse_sort_crs_impl.hpp#L327

If we don't have an appropriate TPL, Kokkos just copies to the host and sorts there, killing performance.

@brian-kelley suggested using the bulk heuristic as a workaround

```c++
#if defined(KOKKOS_ENABLE_HIP) && !defined(KOKKOS_ENABLE_ROCTHRUST)
(void) maxDeg;
(void) avgDeg;
return false;
#else
return (maxDeg / 10 > avgDeg) || (maxDeg > 1024);
#endif
```

**MI300A, rocm 6.1.2, rocThrust 2.0.1 (`THRUST_VERSION=200001`)**
| Configuration | Laplace 3D MueLu level 2 Sort Time |
|-|-|
| `develop` (bulk sort + rocThrust) | 0.00625|
| `develop` (bulk sort) |0.055 |
| `develop` (no bulk sort) | 0.0028s |

Basically, it may make sense to just disable the "bulk sort" (regardless of whether rocThrust is used or not) for MI300A
